lxd-agent/main_agent.go

```diff
@@ -138,26 +147,24 @@ func (c *cmdAgent) Run(cmd *cobra.Command, args []string) error {
 
 	// Check context ID periodically, and restart the HTTP server if needed.
 	go func() {
-		for range time.Tick(30 * time.Second) {
-			cid, err := vsock.ContextID()
-			if err != nil {
-				continue
-			}
-
-			if d.localCID == cid {
+		for {
+			time.Sleep(30 * time.Second)
+			cid, err := waitVsockContextID(context.Background())
```

Comment:
Is there a reason to call `waitVsockContextID` here inside the loop? It means we wait for 30 seconds after a context ID change but after 30 seconds we're checking every second again (for 30 seconds, then repeat).